istioctl/pkg/waypoint/waypoint.go

```diff
@@ -146,6 +151,33 @@ func Cmd(ctx cli.Context) *cobra.Command {
 				}
 				return err
 			}
+			if waitReady {
+				startTime := time.Now()
+				for {
+					programmed := false
+					gwc, err := kubeClient.GatewayAPI().GatewayV1beta1().Gateways(ctx.NamespaceOrDefault(ctx.Namespace())).Get(context.TODO(), gw.Name, metav1.GetOptions{})
+					if err == nil {
+						// Check if gateway has Programmed condition set to true
+						for _, cond := range gwc.Status.Conditions {
+							if cond.Type == string(gatewayv1.GatewayConditionProgrammed) && string(cond.Status) == "True" {
+								programmed = true
+								break
+							}
+						}
+					}
+					if programmed {
+						break
+					}
+					if time.Since(startTime) > waitTimeout {
+						errorMsg := fmt.Sprintf("timed out while waiting for waypoint %v/%v", gw.Namespace, gw.Name)
+						if err != nil {
+							errorMsg += fmt.Sprintf(": %s", err)
+						}
+						return fmt.Errorf(errorMsg)
+					}
+					time.Sleep(1 * time.Second)
```

Comment:
Can we use things like ticker instead of sleep?

Reply:
Changed to use a for loop on ticker channel